components/scenario-screen/scenario-edit.tsx

```diff
@@ -26,38 +28,31 @@ export const ScenarioEdit = () => {
     mode: "onChange",
   });
 
-  // Reset form when scenario changes
   useEffect(() => {
     if (currentScenario) {
-      form.reset({
-        id: currentScenario.id,
-        name: currentScenario.name,
-        overview: {
-          title: currentScenario.overview.title,
-          text: currentScenario.overview.text,
-          image: currentScenario.overview.image,
-        },
-        summary: {
-          title: currentScenario.summary.title,
-          text: currentScenario.summary.text,
-          image: currentScenario.summary.image,
-        },
-        steps: currentScenario.steps,
-      });
+      const formData = scenarioToFormData(currentScenario);
+      form.reset(formData);
     }
   }, [currentScenario, form.reset]);
-
   const onSubmit = (data: ScenarioFormData) => {
     if (selectedScenario === null) return;
 
-    const updatedScenario = {
+    const updatedScenario: Scenario = {
       ...data,
-      steps: currentScenario?.steps || [], // Preserve existing steps
+      overview: {
+        ...data.overview,
+        image: data.overview.image || '',
```

Comment:
nullish coalescing? is see this on some places where i think this would be better suited than the logical or